dashboard/lib/services/script_seed.rb

```diff
@@ -145,8 +145,12 @@ def self.seed_from_json(json_string)
 
         seed_context.script = import_script(script_data)
 
-        # course version must be set before resources are imported.
-        CourseOffering.add_course_offering(seed_context.script)
+        # Course version must be set before resources are imported. If the
+        # script is in a unit group, we must wait and let the next seed step set
+        # the course version on the unit group before resources can be imported.
+        if seed_context.script.is_course
```

Comment:
Does `rake seed:courses` add the course offering to a script?

Reply:
kind of. `rake seed:courses` adds the course offering to a unit group. after that, a call to script.get_course_version will navigate through that unit group to find the course version (and, if it wanted, the course offering).
* https://github.com/code-dot-org/code-dot-org/blob/d01e89082009cad3c14055da5c6f9c5ac66038f3/dashboard/lib/tasks/seed.rake#L216-L218
* https://github.com/code-dot-org/code-dot-org/blob/75c067c90a76c5f8bfff6c99329a0f621e5069b8/dashboard/app/models/unit_group.rb#L92

Reply:
Ahh makes sense. So this branch was never needed for scripts in a unit group?

Reply:
it's more that this codepath was probably totally broken for scripts in a unit group before this PR. I had only been testing on scripts which were standalone courses i.e. `is_course`.